### PR TITLE
date picker on insights

### PR DESCRIPTION
**Pull request by elevchyt**
Added a date picker on insights with **dateFrom** and **dateTo** params that is limited to having its latest date to today.

### CODE REVIEW

File: apps/web/src/app/(authenticated)/insights/components/order-filters.tsx [hunk 1]
```diff
@@ -109,6 +121,29 @@ export default function OrderFilters(): React.ReactNode {
...
+    const newParams = new URLSearchParams(searchParams.toString());
+    const dateFrom = dates[0]?.getTime();
+    const dateTo = dates[1]?.getTime();
+    if (dateFrom && dateTo) {
+      resetInsights();
+      newParams.set(dateFromKey, String(dateFrom));
+      newParams.set(dateToKey, String(dateTo));
+      const newURL = `${pathname}?${newParams.toString()}`;
+      startTransition(() => {
+        router.replace(newURL);
```
Comment (trixobird) [inline apps/web/src/app/(authenticated)/insights/components/order-filters.tsx:134]:
```suggestion
        window.history.replaceState(null, '', newURL)
```
Here are the docs: https://nextjs.org/docs/app/building-your-application/routing/linking-and-navigating#using-the-native-history-api you can replace the url without reloading the page.

Which means that if we have the insights call in a useEffect that is dependent only on searchparams, we should be re-rendering only the changed component and the insights area.

File: apps/web/src/app/(authenticated)/insights/components/order-filters.tsx [hunk 2]
```diff
@@ -179,22 +214,37 @@ export default function OrderFilters(): React.ReactNode {
...
+                ]}
+                value={getIntervalValue()}
+                onChange={handleIntervalChange}
+                allowDeselect={false}
+                comboboxProps={{ transitionProps: { transition: 'fade-down', duration: 200 } }}
+                scrollAreaProps={{ type: 'always', offsetScrollbars: 'y' }}
+                maw={150}
+                disabled={isPending}
+              />
+              <DatePickerInput
```
Comment (trixobird) [inline apps/web/src/app/(authenticated)/insights/components/order-filters.tsx:237]: When I change locale to en-US the week does not start on Sunday

[the remark this answers is not in the document]
Comment (elevchyt) [reply apps/web/src/app/(authenticated)/insights/components/order-filters.tsx:237]: Fixed!

File: apps/web/src/app/(authenticated)/insights/components/order-filters.tsx (hunk 1 above)
Comment (elevchyt) [reply apps/web/src/app/(authenticated)/insights/components/order-filters.tsx:138]: HUGE

Comment (elevchyt) [reply apps/web/src/app/(authenticated)/insights/components/order-filters.tsx:138]: Actually, I tried it on the page size dropdown and seems to be rerendering other stuff too :/
Also, doesn't seem to re-fetch data and shows "no results".

Comment (trixobird) [reply apps/web/src/app/(authenticated)/insights/components/order-filters.tsx:138]: Yes I know there are some issues, but I am assuming it is this neighbouring issue that triggers the re-render. I don't know what would be causing the no re-fetch though. We probably need to test this on a clean project

**Review by trixobird — APPROVED** [review]

**Pull request merged**